src/adapter/src/catalog/open.rs

```diff
@@ -1766,15 +1787,18 @@ fn add_new_builtin_cluster_replicas_migration(
 }
 
 pub(crate) fn builtin_cluster_replica_config(
-    builtin_cluster_replica_size: String,
+    replica_size: (String, Option<&ReplicaAllocation>),
 ) -> mz_catalog::durable::ReplicaConfig {
     mz_catalog::durable::ReplicaConfig {
         location: mz_catalog::durable::ReplicaLocation::Managed {
             availability_zone: None,
             billed_as: None,
-            disk: false,
+            disk: replica_size
+                .1
+                .map(|r| r.disk_limit.is_some())
+                .unwrap_or(false),
             internal: false,
-            size: builtin_cluster_replica_size,
+            size: replica_size.0,
```

Comment:
Is the rest of these hard-coded values correct?

Reply:
uh... yeah, I think we can take the rest of these values, but TBH I don't know what happens if someone were to create a replica that replaces the builtin one. I assume we wouldn't inherit these values, although I'm assuming we'd have to take special care to ensure it's un-billed.

Reply:
This is only used during startup, so if we create a replica on a running instance, then it's configuration is determined from the DDL command itself.

Reply:
ok, this has been simplified quite a bit... I'm just using `is_cluster_size_v2` and keeping much of this code the same as it was.